source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -989,6 +989,10 @@ ServerContextImpl::ServerContextImpl(Stats::Scope& scope,
       });
 
   for (auto& ctx : tls_contexts_) {
+    if (config.requestClientCertificate()) {
+      SSL_CTX_set_verify(ctx.ssl_ctx_.get(), SSL_VERIFY_PEER, nullptr);
```

Comment:
Looks like we already do similar thing in https://github.com/envoyproxy/envoy/blob/master/source/extensions/transport_sockets/tls/context_impl.cc#L112-L122 (this was merged yesterday), so does that fill the actual use case of `request_client_certificate`? cc @jimini-lumox

Reply:
going to test that out, maybe it does. 😄

Reply:
This ignores and removes the flags already set in `ContextImpl::ContextImpl` (regardless of the yesterday's commit).

Reply:
@lizan sorry for the delay, had to test a couple of scenarios. you are correct, https://github.com/envoyproxy/envoy/pull/9172 handles this use case!